pkg/docker/login_test.go

```diff
@@ -0,0 +1,39 @@
+package docker
+
+import (
+	"io/ioutil"
+	"os"
+	"sort"
+	"testing"
+
+	"get.porter.sh/porter/pkg/exec/builder"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+)
+
+func TestMixin_LoginEnv(t *testing.T) {
+	//login test
+	os.Setenv("DOCKER_USERNAME", "gmadhok")
+	os.Setenv("DOCKER_PASSWORD", "password")
+	b, err := ioutil.ReadFile("testdata/login-input.yaml")
+	require.NoError(t, err)
+
+	var action Action
+	err = yaml.Unmarshal(b, &action)
+	require.NoError(t, err)
+	require.Len(t, action.Steps, 1)
+
+	step := action.Steps[0]
+	assert.Equal(t, "Login to docker", step.Description)
+
+	args := step.GetArguments()
+	assert.Equal(t, "login", args[0])
+
+	flags := step.GetFlags()
+	sort.Sort(flags)
+	assert.Equal(t, builder.Flags{builder.NewFlag("p", "password"), builder.NewFlag("u", "gmadhok")}, flags)
+	os.Unsetenv("DOCKER_USERNAME")
```

Comment:
A potential problem here is that if for any reason the test fails and doesn't make it to the end where you call `os.Unsetenv` then the environment variables will remain set. Then in subsequent tests where you probably didn't expect those variables to be set, they could mess up those tests too. 

In Go, when we have code that needs to be executed at the end of the function, no matter what, you can use the `defer` keyword. In this case, you can do this

```golang
os.Setenv("DOCKER_PASSWORD", "password")
defer func(){
  os.Unsetenv("DOCKER_USERNAME")
  os.Unsetenv("DOCKER_PASSWORD")
}()
```

In this case I'm using an anonymous function so that I can do multiple things. If I just had needed to do one thing, I could have just done `defer os.Unsetenv("DOCKER_USERNAME")`

Reply:
Got it - just fixed that